ui/angular_components.md

```diff
@@ -0,0 +1,25 @@
+### Angular Component
+
+Location: `app/assets/javascripts/components`. 
+
+#### Basic pattern
+A component shouldn't modify any data or DOM that's out of its scope. All communication should go via bindings. One-way binding should be preferred over two-way binding.  
+Set `controllerAs: "vm"`.  
+`templateUrl:` should be prefer over `template:` but at the moment it's impossible to test that component's HTML is generated correctly if `templateUrl:` is used. So temporarily `template:` should be used instead of `templateUrl:`. 
+
+#### Existing components
+
+`miq-button`
```

Comment:
Probably better as `##### miq-button` .. so that it's clear a section starts here.